components/director/internal/systemfetcher/service.go

```diff
@@ -219,29 +223,50 @@ func (s *SystemFetcher) processSystemsForTenant(ctx context.Context, tenantMappi
 		}
 	}
 
-	err = tx.Commit()
-	if err != nil {
+	if err = tx.Commit(); err != nil {
 		return errors.Wrap(err, fmt.Sprintf("failed to commit applications for tenant %s", tenantMapping.ExternalTenant))
 	}
 
 	return nil
 }
 
-func (s *SystemFetcher) convertSystemToAppRegisterInput(sc System) (model.ApplicationRegisterInput, string) {
+func (s *SystemFetcher) convertSystemToAppRegisterInput(ctx context.Context, sc System) (*model.ApplicationRegisterInputWithTemplate, error) {
+	input, err := s.appRegisterInput(ctx, sc)
+	if err != nil {
+		return nil, err
+	}
+
+	return enrichAppRegisterInput(*input, sc), nil
+}
+
+func (s *SystemFetcher) appRegisterInput(ctx context.Context, sc System) (*model.ApplicationRegisterInput, error) {
+	if len(sc.TemplateID) > 0 {
+		return s.templatedApplicationResolver.ApplicationRegisterInputFromTemplate(ctx, sc)
+	}
+
+	return &model.ApplicationRegisterInput{
+		Name:         sc.DisplayName,
+		Description:  &sc.ProductDescription,
+		ProviderName: &sc.InfrastructureProvider,
+		BaseURL:      &sc.BaseURL,
+		SystemNumber: &sc.SystemNumber,
+	}, nil
+}
+
+// enrichAppRegisterInput replaces the base part of the application with all the information that could be retrieved from the system fetcher.
+func enrichAppRegisterInput(input model.ApplicationRegisterInput, sc System) *model.ApplicationRegisterInputWithTemplate {
 	initStatusCond := model.ApplicationStatusConditionInitial
-	baseURL := sc.BaseURL
-
-	appRegisterInput := model.ApplicationRegisterInput{
-		Name:            sc.DisplayName,
-		Description:     &sc.ProductDescription,
-		BaseURL:         &baseURL,
-		ProviderName:    &sc.InfrastructureProvider,
-		StatusCondition: &initStatusCond,
-		SystemNumber:    &sc.SystemNumber,
-		Labels: map[string]interface{}{
-			"managed": "true",
-		},
+
+	input.StatusCondition = &initStatusCond
+	input.SystemNumber = &sc.SystemNumber
+
+	if len(input.Labels) == 0 {
+		input.Labels = make(map[string]interface{}, 1)
 	}
+	input.Labels["managed"] = "true"
```

Comment:
You can add the managed label to the template input override and make use of the enrichLabels functionality.

Maybe the same applies for the StatusCondition and the SystemNumber as well and get rid of this function entirely.